src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerProcessor.cs

```diff
@@ -35,59 +73,130 @@ public ConsoleLoggerProcessor(IConsole console, IConsole errorConsole)
 
         public virtual void EnqueueMessage(LogMessageEntry message)
         {
-            if (!_messageQueue.IsAddingCompleted)
-            {
-                try
-                {
-                    _messageQueue.Add(message);
-                    return;
-                }
-                catch (InvalidOperationException) { }
-            }
-
-            // Adding is completed so just log the message
-            try
+            // cannot enqueue when adding is completed
+            if (!Enqueue(message))
             {
                 WriteMessage(message);
             }
-            catch (Exception) { }
         }
 
         // for testing
         internal void WriteMessage(LogMessageEntry entry)
         {
-            IConsole console = entry.LogAsError ? ErrorConsole : Console;
-            console.Write(entry.Message);
+            try
+            {
+                IConsole console = entry.LogAsError ? ErrorConsole : Console;
+                console.Write(entry.Message);
+            }
+            catch (Exception)
+            {
+                CompleteAdding();
+            }
         }
 
         private void ProcessLogQueue()
         {
-            try
+            while (!_isAddingCompleted || _messageQueue.Count > 0)
```

Comment:
I don't understand this code.  This seems both very thread-unsafe and unnecessary.  _messageQueue should only be accessed while holding the lock, and doesn't TryDequeue already do all the right checks?  Why isn't this just `while (TryDequeue(...))` rather than having TryDequeue be inside the if block?